framework/auth/views.py

```diff
@@ -198,88 +198,154 @@ def forgot_password_post(auth, **kwargs):
     return {}
 
 
-@collect_auth
-def auth_login(auth, **kwargs):
+def login_and_register_handler(auth, login=True, campaign=None, service_url=None, logout=None, **kwargs):
```

Comment:
[To CR] Login/Register logic is now separated from the view. Add this helper function `login_and_register_handler` do most of the flow decisions and return a data object containing actions for `auth_login` and `auth_register`.